internal/buffer/save.go

```diff
@@ -25,10 +24,46 @@ import (
 // because hashing is too slow
 const LargeFileThreshold = 50000
 
-// overwriteFile opens the given file for writing, truncating if one exists, and then calls
-// the supplied function with the file as io.Writer object, also making sure the file is
-// closed afterwards.
-func overwriteFile(name string, enc encoding.Encoding, fn func(io.Writer) error, withSudo bool) (err error) {
+func (b *Buffer) fileWriter(file io.Writer) (int, error) {
```

Comment:
`writeFile()`?

"writer" and the like is how we Go-idiomatically name interfaces, not functions.